pkgs/development/libraries/libvpx/default.nix

```
@@ -164,7 +168,7 @@ stdenv.mkDerivation rec {
                     experimentalFpMbStatsSupport ||
                     experimentalEmulateHardwareSupport) "experimental")
   ] ++ optionals (stdenv.isBSD || stdenv.hostPlatform != stdenv.buildPlatform) [
-    "--force-target=${stdenv.hostPlatform.parsed.cpu.name}-${kernel}-gcc"
+    "--target=${cpu}-${kernel}-gcc"
```

```suggestion
    "--force-target=${cpu}-${kernel}-gcc"
```

the change from `--force-target` to `--target` seems to break `nix-build -A pkgsCross.aarch64-multiplatform.libvpx`. if i use `--force-target` it builds for aarch64 and `pkgsCross.armv7l-hf-multiplatform.libvpx`. i haven't checked what the difference is, the error was during configure:
```
Unrecognized toolchain 'aarch64-linux-gcc'
```

---

`aarch64-linux-gnu` isn't a valid target, apparently it is `arm64`.

---

it's not just aarch64, unfortunately. somehow all these systems work on master (via `pkgsCross.$system.libvpx`), but break with this patch (not exhaustive):
- gnu32
- mips-linux-gnu
- mipsel-linux-gnu
- musl32
- musl64
- ppc64
- raspberryPi (armv6l)
- riscv32
- riscv64
- sheevaplug  (armv5tel)

---

Interesting, looking at the supported systems according to the configure script. These are the only supported targets:
```
Supported targets:
    arm64-android-gcc        arm64-darwin-gcc         arm64-darwin20-gcc
    arm64-darwin21-gcc       arm64-darwin22-gcc       arm64-darwin23-gcc
    arm64-linux-gcc          arm64-win64-gcc          arm64-win64-vs15
    arm64-win64-vs16         arm64-win64-vs16-clangcl arm64-win64-vs17
    arm64-win64-vs17-clangcl
    armv7-android-gcc        armv7-darwin-gcc         armv7-linux-rvct
    armv7-linux-gcc          armv7-none-rvct          armv7-win32-gcc
    armv7-win32-vs14         armv7-win32-vs15         armv7-win32-vs16
    armv7-win32-vs17
    armv7s-darwin-gcc
    armv8-linux-gcc
    loongarch32-linux-gcc
    loongarch64-linux-gcc
    mips32-linux-gcc
    mips64-linux-gcc
    ppc64le-linux-gcc
    sparc-solaris-gcc
    x86-android-gcc          x86-darwin8-gcc          x86-darwin8-icc
    x86-darwin9-gcc          x86-darwin9-icc          x86-darwin10-gcc
    x86-darwin11-gcc         x86-darwin12-gcc         x86-darwin13-gcc
    x86-darwin14-gcc         x86-darwin15-gcc         x86-darwin16-gcc
    x86-darwin17-gcc         x86-iphonesimulator-gcc  x86-linux-gcc
    x86-linux-icc            x86-os2-gcc              x86-solaris-gcc
    x86-win32-gcc            x86-win32-vs14           x86-win32-vs15
    x86-win32-vs16           x86-win32-vs17
    x86_64-android-gcc       x86_64-darwin9-gcc       x86_64-darwin10-gcc
    x86_64-darwin11-gcc      x86_64-darwin12-gcc      x86_64-darwin13-gcc
    x86_64-darwin14-gcc      x86_64-darwin15-gcc      x86_64-darwin16-gcc
    x86_64-darwin17-gcc      x86_64-darwin18-gcc      x86_64-darwin19-gcc
    x86_64-darwin20-gcc      x86_64-darwin21-gcc      x86_64-darwin22-gcc
    x86_64-darwin23-gcc      x86_64-iphonesimulator-gcc x86_64-linux-gcc
    x86_64-linux-icc         x86_64-solaris-gcc       x86_64-win64-gcc
    x86_64-win64-vs14        x86_64-win64-vs15        x86_64-win64-vs16
    x86_64-win64-vs17
    generic-gnu
```

---

i think we can either use `--target` and map to libvpx's own (bespoke?) database of target triples, or we can use `--force-target` and use the nixpkgs flavor of triples.

- [here's](https://github.com/webmproject/libvpx/blob/main/configure#L98) their target database
- [here's](https://github.com/webmproject/libvpx/blob/main/build/make/configure.sh#L869) the primary way in which `--force-target` differs from ordinary `--target` (at a glance)

---

Ideally, we should be using `--target` exclusively. It sounds like `--force-target` is like `--target` except it bypasses the toolchain check.

---

i'd guess they do some feature guessing (like e.g. neon fpu) based on the target, which seems to be what caused the original aarch64-multiplatform failure.

i don't see armv5tel (sheevaplug) or armv6l (raspberryPi) in their platform list though, so we may still need to `--force-target` for nixpkgs stdenv triples which can't be mapped onto that list you shared.

---

I've did some changes that I'll push here in a minute. I've tuned it to where there's an `isGeneric` variable that determines if we should be using `generic-gnu`. On all of the other cases, we will be using the specific target.